code/datums/ai/ai.dm

```diff
@@ -1,5 +1,35 @@
 /datum/ai
 	var/name
+	var/mob/living/body		// The parent mob we control.
+	var/wait_for = 	0 		// The next time we can process.
+	var/run_interval = 1 	// How long to wait between processes.
+	
+/datum/ai/New(var/mob/living/body)
+	src.body = body
```

Comment:
Aiee var shadowing. Can you name the arg `new_body` or something rather than relying on `src.`?